args.go

```diff
@@ -162,7 +162,7 @@ func (args *argsReader) Close() error {
 
 		err := args.dec.Err()
 
-		if args.done != nil {
+		if args.done != nil && err != nil {
```

Comment:
Nope, we have to notify the `done` channel even if there were no errors, otherwise each connection will only be able to handle one request, then will timeout.